src/d3-cloud.directive.js

```diff
@@ -37,173 +37,191 @@
-
-        $scope.createCloud = function(words) {
-          var cloudWidth = $element[0].clientWidth + 0;
-          var cloudHeight = $element[0].clientWidth + 0;
-
-          var minScore = 0;
-          var maxScore = 1;
-          var slope = 1;
-
-          words.map(function(d) {
-            if (minScore > d.score) {
-              minScore = d.score;
-            }
-            if (maxScore < d.score) {
-              maxScore = d.score;
-            }
-          });
-
-          if (maxScore !== minScore) {
-            slope = slopeFactor / (maxScore - minScore);
-          }
-
-          $scope.cloud = d3.layout.cloud().size([cloudWidth, cloudHeight]);
-          $scope.cloud
-            .words(words.map(function(d) {
-              return {
-                text: d.name,
-                size: d.score * slope + slopeBase
-              };
-            }))
-            .padding(padding)
-            .rotate(rotate)
-            .font($scope.font)
-            .fontSize(function(d) {
-              return d.size;
-            })
-            .on('end', draw)
-            .start();
-        };
+      link: function ($scope, $element, $attrs) {
+        angular.element(document).ready(function () {
+          $scope.events = $scope.events || {};
+          $scope.font = $scope.font || 'Impact';
+          $scope.filterFunction = $scope.filterFunction || function (words) {
+            return words;
+          };
+          var padding = $attrs.padding ? Number($scope.padding) : 5;
+          var rotate = $scope.rotate && function (d, i) {
+              return $scope.rotate({word: $scope.words[i]});
+            } || function () {
+              return ~~(Math.random() * 2) * 90 - 45;
+            };
+          var slopeBase = $attrs.slopeBase ? Number($scope.slopeBase) : 2;
+          var slopeFactor = $attrs.slopeFactor ? Number($scope.slopeFactor) : 30;
+
+          $scope.createCloud = function (words) {
+            var cloudWidth = $element[0].clientWidth + 0;
+            var cloudHeight = $element[0].clientHeight + 0;
+            var minScore = 0;
+            var maxScore = 1;
+            var slope = 1;
+
+            words.map(function (d) {
+              if (minScore > d.score) {
+                minScore = d.score;
+              }
+              if (maxScore < d.score) {
+                maxScore = d.score;
+              }
+            });
 
-        $scope.updateCloud = function(words) {
-          var cloudWidth = $element[0].clientWidth + 0;
-          var cloudHeight = $element[0].clientWidth + 0;
+            if (maxScore !== minScore) {
+              slope = slopeFactor / (maxScore - minScore);
+            }
 
-          var minScore = 0;
-          var maxScore = 1;
-          var slope = 1;
+            $scope.cloud = d3.layout.cloud().size([cloudWidth, cloudHeight]);
+            $scope.cloud
+              .words(words.map(function (d) {
+                const result = {
```

Comment:
Don't use `const` here, just use `var`. Better legacy support, and result is not used as constant..